flict/flictlib/translator.py

```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
-                translations_map[item['value']] = item
-
-        return {
-            'original': translations_object,
-            'translations_map': translations_map
-        }
-
-
 def read_translations(translations_file):
     symbols = {}
-    with open(translations_file) as fp:
-        translations_object = json.load(fp)
-        translations = translations_object["translations"]
-        for item in translations:
-            #print(" let's check...." + str(item), end="")
-            if 'license_expression' in item and 'spdx_id' in item:
-                le = item['license_expression'].lower()
-                if "scancode" in le:
-                    logger.main_logger.debug(" IGNORING since scancode")
-                    pass
-                else:
-                    #print(" OK ")
-                    transl = item['spdx_id']
-                    key = item['license_expression']
-                    if transl not in symbols:
-                        symbols[transl] = []
-                    symbols[transl].append(key)
+    try:
+        with open(translations_file) as file_:
+            translations_object = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading translations file failed")
+
+    translations = translations_object["translations"]
+    translations_map = {item['value']: item
+                        for item in translations
+                        if item.get('translation', False)}
+
+    for item in translations:
+        if 'license_expression' in item and 'spdx_id' in item:
+            le = item['license_expression'].lower()
+            if "scancode" in le:
+                logger.main_logger.debug(" IGNORING since scancode")
             else:
-                raise Exception("Failed parsing item:  " + str(item))
+                transl = item['spdx_id']
+                key = item['license_expression']
+                if transl not in symbols:
+                    symbols[transl] = []
+                symbols[transl].append(key)
+        else:
+            raise RuntimeError(f"Failed parsing item: {item}")
+    results = {'original': translations_object,
+               'translations_map': translations_map,
+               'symbols': symbols}
+    # enabling this crashes lots of tests :)
+    # return results
+    read_translations.results = results
     return symbols
 
 
 def read_packages_file(jsonfile, translations):
-    with open(jsonfile) as fp:
-        packages = json.load(fp)
-        # TODO: sync with flict (should be "package")
-        package = packages["component"]
-        deps = package["dependencies"]
-        license = package["license"].strip()
-        package["license"] = update_license(translations, license)
-        update_packages(translations, deps)
-        return packages
+    try:
+        with open(jsonfile) as file_:
+            packages = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
```

Comment:
I personally would also cover PermissionError here in case the user doesn't have read access to the file